pandas/tests/frame/test_analytics.py

```diff
@@ -543,107 +537,124 @@ def test_stat_operators_attempt_obj_array(self, method):
-        self.tsframe.loc[15:, 2] = nan
+    def test_cummin(self, datetime_frame):
+        datetime_frame.loc[5:10, 0] = nan
+        datetime_frame.loc[10:15, 1] = nan
+        datetime_frame.loc[15:, 2] = nan
 
         # axis = 0
-        cummin = self.tsframe.cummin()
-        expected = self.tsframe.apply(Series.cummin)
+        cummin = datetime_frame.cummin()
+        expected = datetime_frame.apply(Series.cummin)
         tm.assert_frame_equal(cummin, expected)
 
         # axis = 1
-        cummin = self.tsframe.cummin(axis=1)
-        expected = self.tsframe.apply(Series.cummin, axis=1)
+        cummin = datetime_frame.cummin(axis=1)
+        expected = datetime_frame.apply(Series.cummin, axis=1)
         tm.assert_frame_equal(cummin, expected)
 
         # it works
         df = DataFrame({'A': np.arange(20)}, index=np.arange(20))
         result = df.cummin()  # noqa
 
         # fix issue
-        cummin_xs = self.tsframe.cummin(axis=1)
-        assert np.shape(cummin_xs) == np.shape(self.tsframe)
+        cummin_xs = datetime_frame.cummin(axis=1)
+        assert np.shape(cummin_xs) == np.shape(datetime_frame)
 
-    def test_cummax(self):
-        self.tsframe.loc[5:10, 0] = nan
-        self.tsframe.loc[10:15, 1] = nan
-        self.tsframe.loc[15:, 2] = nan
+    def test_cummax(self, datetime_frame):
+        datetime_frame.loc[5:10, 0] = nan
+        datetime_frame.loc[10:15, 1] = nan
+        datetime_frame.loc[15:, 2] = nan
 
         # axis = 0
-        cummax = self.tsframe.cummax()
-        expected = self.tsframe.apply(Series.cummax)
+        cummax = datetime_frame.cummax()
+        expected = datetime_frame.apply(Series.cummax)
         tm.assert_frame_equal(cummax, expected)
 
         # axis = 1
-        cummax = self.tsframe.cummax(axis=1)
-        expected = self.tsframe.apply(Series.cummax, axis=1)
+        cummax = datetime_frame.cummax(axis=1)
+        expected = datetime_frame.apply(Series.cummax, axis=1)
         tm.assert_frame_equal(cummax, expected)
 
         # it works
         df = DataFrame({'A': np.arange(20)}, index=np.arange(20))
         result = df.cummax()  # noqa
 
         # fix issue
-        cummax_xs = self.tsframe.cummax(axis=1)
-        assert np.shape(cummax_xs) == np.shape(self.tsframe)
+        cummax_xs = datetime_frame.cummax(axis=1)
+        assert np.shape(cummax_xs) == np.shape(datetime_frame)
 
-    def test_max(self):
+    def test_max(self, float_frame_with_na, int_frame,
+                 float_frame, float_string_frame):
         with warnings.catch_warnings(record=True):
-            self._check_stat_op('max', np.max, check_dates=True)
-        self._check_stat_op('max', np.max, frame=self.intframe)
+            self._check_stat_op('max', np.max, float_frame_with_na,
+                                float_frame, float_string_frame,
+                                check_dates=True)
+        self._check_stat_op('max', np.max, int_frame, float_frame,
```

Comment:
In tests like this one, all the parts of `_check_stat_op` that depend on `float_frame` and `float_string_frame`  get tested twice unnecessarily

Reply:
hmm, can you have a look at the git history and see if you can figure out why. otherwise ok to remove.